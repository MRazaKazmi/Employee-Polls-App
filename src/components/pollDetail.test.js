import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import PollDetail from './pollDetail';
import '@testing-library/jest-dom/extend-expect';

const mockStore = configureStore([]);

describe('PollDetail', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      authedUser: 'sarahedo',
      users: {
        sarahedo: {
          id: 'sarahedo',
          name: 'Sarah Edo',
          avatarURL: '/path/to/avatar.jpg',
          answers: {
            "8xf0y6ziyjabvozdd253nd": 'optionOne',
          },
          questions: ['8xf0y6ziyjabvozdd253nd']
        },
        tylermcginnis: {
          id: 'tylermcginnis',
          name: 'Tyler McGinnis',
          avatarURL: '/path/to/tyler.jpg',
          answers: {},
          questions: []
        }
      },
      questions: {
        "8xf0y6ziyjabvozdd253nd": {
          id: '8xf0y6ziyjabvozdd253nd',
          author: 'tylermcginnis',
          timestamp: 1467166872634,
          optionOne: {
            votes: ['sarahedo'],
            text: 'Build our new application with Javascript',
          },
          optionTwo: {
            votes: [],
            text: 'Build our new application with Python',
          }
        }
      }
    });
  });

  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/questions/8xf0y6ziyjabvozdd253nd']}>
          <Routes>
            <Route path="/questions/:id" element={<PollDetail />} />
          </Routes>
        </MemoryRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot when question is not found', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/questions/unknown']}>
          <Routes>
            <Route path="/questions/:id" element={<PollDetail />} />
          </Routes>
        </MemoryRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
