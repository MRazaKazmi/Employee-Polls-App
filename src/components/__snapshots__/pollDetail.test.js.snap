// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollDetail matches the snapshot 1`] = `
<div>
  <div
    class="poll-detail-container"
  >
    <h2>
      Would you rather...
    </h2>
    <div
      class="author-info"
    >
      <img
        alt="Author Avatar"
        class="avatar"
        src="/path/to/tyler.jpg"
      />
      <p>
        Question by: 
        Tyler McGinnis
      </p>
    </div>
    <div
      class="poll-options"
    >
      <div
        class="option"
      >
        <button>
          Build our new application with Javascript
        </button>
      </div>
      <div
        class="option"
      >
        <button>
          Build our new application with Python
        </button>
      </div>
    </div>
    <p>
      Your answer: 
      optionOne
    </p>
  </div>
</div>
`;

exports[`PollDetail matches the snapshot when question is not found 1`] = `
<div>
  <p>
    Question not found.
  </p>
</div>
`;
